LibTest/js_interop/importModule_A01_t01.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion JSPromise<JSObject> importModule( JSAny moduleName )
+///
+/// Dynamically imports a JavaScript module with the given `moduleName` using
+/// the JavaScript `import()` syntax.
+///
+/// See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import
+/// for more details.
+///
+/// Returns a [JSPromise] that resolves to a [JSObject] that's the module
+/// namespace object.
+///
+/// @description Check that [importModule] imports a JavaScript module with the
+/// given `moduleName` and returns a [JSPromise] that resolves to a [JSObject]
+/// that's the module namespace object.
+/// @author sgrekhov22@gmail.com
+
+import 'dart:js_interop';
+import '../../Utils/expect.dart';
+
+extension type Module(JSObject _) implements JSObject {
+  external int answer();
+}
+
+extension type A._(JSObject _) implements JSObject {
+  external A(id, name);
```

Comment:
I believe the lack of types here will be a compile-time error as `dynamic` is not allowed.